test/specs/integration.spec.js

```diff
@@ -65,125 +62,116 @@ describe('Integration tests', function() {
         cm1.replaceRange('', {line: 0, ch: 10}, {line: 0, ch: 13});
         cm1.replaceRange('', {line: 0, ch: 0},  {line: 0, ch: 3});
       });
+      setTimeout(function() {
+        expect(cm2.getValue()).toEqual('3456789');
+        done();
+      }, 500);
     });
-    waitsFor(function() { return cm2.getValue() === '3456789' }, 'Got correct end value.');
   });
 
-  it('Random text changes', function () {
+  it('Random text changes', function (done) {
     var ref = new Firebase('https://firepad-test.firebaseio-demo.com').push();
-    var cm1 = CodeMirror(cmDiv());
-    var cm2 = CodeMirror(cmDiv());
+    var cm1 = CodeMirror(hiddenDiv());
+    var cm2 = CodeMirror(hiddenDiv());
     var firepad1 = new Firepad(ref, cm1);
     var firepad2 = new Firepad(ref, cm2);
 
-    var n = 25;
-
-    var done = false;
-    function step () {
-      if (n == 0) {
-        done = true;
-        return;
+    function step (times) {
+      if (times == 0) {
+        done();
+      }else{
+        randomOperation(cm1);
+        setTimeout(function() {
```

Comment:
Sure, that'd be fine with me.  It's not pretty, but it's equivalent of what jasmine 1 was doing. :-)